### PR TITLE
planner: update document about non-prepared plan cache

### PR DESCRIPTION
**Pull request by qw4990**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/14654
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by chrysan — APPROVED** [review]

File: system-variables.md
Comment (Oreoxmt) [inline system-variables.md:1]: @qw4990 Please remove the warning for `tidb_enable_non_prepared_plan_cache_for_dml` and confirm whether the default value should be changed from `OFF` to `ON`?

https://github.com/pingcap/docs-cn/blob/92bcee66abff1af4a83d2b83f16459a75781ecb8/system-variables.md?plain=1#L1637-L1647

File: sql-non-prepared-plan-cache.md [hunk 1]
```diff
@@ -5,12 +5,10 @@ summary: 介绍 TiDB 中非 Prepare 语句执行计划缓存的原理、使用
 
 # 非 Prepare 语句执行计划缓存
 
-> **警告：**
->
-> 非 Prepare 语句执行计划缓存 (Non-Prepared Plan Cache) 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
-
 对于某些非 `PREPARE` 语句，TiDB 可以像 [`Prepare`/`Execute` 语句](/sql-prepared-plan-cache.md)一样支持执行计划缓存。这可以让这些语句跳过优化器阶段，以提升性能。
 
+打开此功能可能会带来一些额外的内存和 CPU 开销，并不一定适用于所有场景，请根据具体的使用情况决定是否打开此项功能。具体见[性能收益](#性能收益)、[内存监控](#监控)。
```
Comment (Oreoxmt) [inline sql-non-prepared-plan-cache.md:10]:
```suggestion
开启非 Prepare 语句执行计划缓存可能会带来额外的内存和 CPU 开销，并不一定适用于所有场景。建议参考[性能收益](#性能收益)和[内存监控](#监控)章节，根据具体的使用情况决定是否开启该功能。
```

File: sql-non-prepared-plan-cache.md [hunk 2]
```diff
@@ -97,7 +95,9 @@ TiDB 对参数化后形式相同的查询，只能缓存一个计划。例如，
 
 ## 性能收益
 
-在内部测试中，开启 Non-Prepared Plan Cache 功能在大多数 TP 场景下可以获得显著的性能收益。但是它也有代价，其自身也有一些额外的性能开销，包括判断查询是否支持、对查询进行参数化等。如果此功能不支持负载中的大多数查询，开启此功能反而可能影响性能。
+在内部测试中，开启 Non-Prepared Plan Cache 功能在大多数 TP 场景下可以获得显著的性能收益。在内部测试中，TPCC 能获得约 4% 的提升，一些 Bank 负载上能获得 10+% 的提升，在 Sysbench RangeScan 上能获得 15% 的提升。
+
+但是这个功能本身也会带来一些额外的 CPU 和内存开销，包括判断查询是否支持、对查询进行参数化、在 Plan Cache 中进行搜索等。如果负载中的多数查询无法被 Cache 命中，开启此功能反而可能影响性能。
```
Comment (Oreoxmt) [inline sql-non-prepared-plan-cache.md:100]:
```suggestion
在内部测试中，开启 Non-Prepared Plan Cache 功能在大多数 TP 场景下可以获得显著的性能收益。例如：TPCC 测试中性能提升约 4%，一些 Bank 负载上提升超过 10%，在 Sysbench RangeScan 上提升达到 15%。

但是这个功能本身也会带来额外的 CPU 和内存开销，包括判断查询是否支持、对查询进行参数化、在 Plan Cache 中进行搜索等。如果负载中的多数查询无法被 Cache 命中，开启此功能反而可能影响性能。
```

File: system-variables.md [hunk 3]
```diff
@@ -1627,15 +1627,12 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
...
-> **警告：**
->
-> 非 Prepare 语句执行计划缓存目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
-
 - 作用域：SESSION | GLOBAL
 - 是否持久化到集群：是
 - 类型：布尔型
 - 默认值：`ON`
 - 这个变量用来控制是否开启[非 Prepare 语句执行计划缓存](/sql-non-prepared-plan-cache.md)。
+- 打开此功能可能会带来一些额外的内存和 CPU 开销，并不一定适用于所有场景，请根据具体的使用情况决定是否打开此项功能。
```
Comment (Oreoxmt) [inline system-variables.md:1635]:
```suggestion
- 开启此功能可能会带来额外的内存和 CPU 开销，并不一定适用于所有场景，请根据具体的使用情况决定是否开启该功能。
```

File: system-variables.md
Comment (qw4990) [reply system-variables.md:1]: `non-prep for dml` is still experimental.

Comment (Oreoxmt) [reply system-variables.md:1]: @qw4990 Please update the warning to 非 Prepare 语句执行计划缓存**支持 DML 语句**目前为实验特性

File: sql-prepared-plan-cache.md [hunk 4]
```diff
@@ -32,6 +32,9 @@ TiDB 优化器对这两类查询的处理是一样的：`Prepare` 时将参数
 - 用参数进行 `int` 和 `string` 比较的查询，如 `c_int >= ?` 或者 `c_int in (?, ?)`等，其中 `?` 为字符串类型，如 `set @x='123'`；此时为了保证结果和 MySQL 兼容性，需要每次对参数进行调整，故不会缓存；
 - 会访问 `TiFlash` 的计划不会被缓存；
 - 大部分情况下计划中含有 `TableDual` 的计划将将不会被缓存，除非当前执行的 `Prepare` 语句不含参数，则对应的 `TableDual` 计划可以被缓存。
+- 访问 TiDB 系统视图的查询，如 `information_schema.columns`，并且建议不使用 `Prepare`/`Execute` 语句来访问系统视图。
```
Comment (Oreoxmt) [inline sql-prepared-plan-cache.md:35]:
```suggestion
- 访问 TiDB 系统视图的查询，如 `information_schema.columns`。不建议使用 `Prepare`/`Execute` 语句访问系统视图。
```

File: system-variables.md
Comment (qw4990) [reply system-variables.md:1]: 已经更新

**Review by Oreoxmt — APPROVED** [review]

File: system-variables.md [hunk 5]
```diff
@@ -1627,21 +1627,18 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
...
 - 默认值：`ON`
 - 这个变量用来控制是否开启[非 Prepare 语句执行计划缓存](/sql-non-prepared-plan-cache.md)。
+- 开启此功能可能会带来额外的内存和 CPU 开销，并不一定适用于所有场景，请根据具体的使用情况决定是否开启该功能。
 
 ### `tidb_enable_non_prepared_plan_cache_for_dml` <span class="version-mark">从 v7.1.0 版本开始引入</span>
 
 > **警告：**
 >
-> 非 Prepare 语句执行计划缓存目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+> 非 Prepare 语句执行计划缓存支持 DML 语句目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
```
Comment (ran-huang) [inline system-variables.md:1641]:
```suggestion
> 针对 DML 语句的非 Prepare 语句执行计划缓存目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
```

File: optimizer-fix-controls.md [hunk 6]
```diff
@@ -57,3 +57,10 @@ SET SESSION tidb_opt_fix_control = '44262:ON,44389:ON';
 - 在某些场景下，当 `IndexJoin` 算子的 `Probe` 端包含 `Selection` 算子时，TiDB 会严重高估 `IndexScan` 的行数，导致在 `IndexJoin` 更好的时候选择了其它的执行计划。
 - TiDB 已经引入了缓解这类问题的改进逻辑。但是由于潜在的计划回退风险，该改进并没有被默认启用。
 - 此开关控制是否启用这个改进。
+
+### [`45132`](https://github.com/pingcap/tidb/issues/45132) <span class="version-mark">从 v7.4.0 版本开始引入</span>
```
Comment (qw4990) [inline optimizer-fix-controls.md:61]: 一个新的改动，也麻烦帮忙看看 @Oreoxmt

File: sql-prepared-plan-cache.md [hunk 7]
```diff
@@ -22,7 +22,7 @@ TiDB 优化器对这两类查询的处理是一样的：`Prepare` 时将参数
 - 访问分区表、临时表或访问表中包含生成列的查询；
 - 查询中包含非关联子查询，例如 `SELECT * FROM t1 WHERE t1.a > (SELECT 1 FROM t2 WHERE t2.b < 1)`；
 - 执行计划中带有 `PhysicalApply` 算子的关联子查询，例如 `SELECT * FROM t1 WHERE t1.a > (SELECT a FROM t2 WHERE t1.b > t2.b)`；
-- 包含 `ignore_plan_cache` 这一 Hint 的查询，例如 `select /*+ ignore_plan_cache() */ * from t`；
+- 包含 `ignore_plan_cache` 或 `set_var` 这两 Hint 的查询，例如 `select /*+ ignore_plan_cache() */ * from t` 或 `select /*+ set_var(max_execution_time=1) */ * from t`；
```
Comment (Oreoxmt) [inline sql-prepared-plan-cache.md:25]:
```suggestion
- 包含 `ignore_plan_cache` 或 `set_var` 这两个 Hint 的查询，例如 `SELECT /*+ ignore_plan_cache() */ * FROM t` 或 `SELECT /*+ set_var(max_execution_time=1) */ * FROM t`；
```

File: optimizer-fix-controls.md [hunk 8]
```diff
@@ -57,3 +57,10 @@ SET SESSION tidb_opt_fix_control = '44262:ON,44389:ON';
...
+
+- 默认值：`1000`
+- 可选值：`[0, 2147483647]`
+- 此开关控制优化器进行启发式访问路径选择的阈值，当某个访问路径（如 `Index_A`）的估算行数远小于其他访问路径时（默认为 `1000` 倍），则跳过代价比较直接选择 `Index_A`。
```
Comment (Oreoxmt) [inline optimizer-fix-controls.md:65]:
```suggestion
- 此开关控制优化器进行启发式访问路径选择的阈值。当某个访问路径（如 `Index_A`）的估算行数远小于其他访问路径时（默认为 `1000` 倍），优化器会跳过代价比较直接选择 `Index_A`。
```

**Review by Oreoxmt — APPROVED** [review]

**Review by ran-huang — APPROVED** [review]

File: system-variables.md [hunk 9]
```diff
@@ -1627,21 +1627,18 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
...
 ### `tidb_enable_non_prepared_plan_cache`
 
-> **警告：**
->
-> 非 Prepare 语句执行计划缓存目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
-
 - 作用域：SESSION | GLOBAL
 - 是否持久化到集群：是
 - 类型：布尔型
 - 默认值：`ON`
```
Comment (qw4990) [inline system-variables.md:1633]:
```suggestion
- 默认值：`OFF`
```

**Review by Oreoxmt — APPROVED** [review]

**Pull request merged**